tobac/utils/general.py

```diff
@@ -631,6 +631,228 @@ def combine_feature_dataframes(
+
+        # Find which labels exist in features for output:
+        index_in_features = np.isin(index, features[id_column])
+
+        # set negative markers to 0 as they are unsegmented
+        labels[labels < 0] = 0
+        bins = np.cumsum(np.bincount(labels.ravel()))
+        argsorted = np.argsort(labels.ravel())
+
+        # apply each function given per func_dict for the labeled regions sorted in ascending order
+        for stats_name in func_dict.keys():
+            # initiate new column in feature dataframe if it does not already exist
+            if stats_name not in features.columns:
+                features[stats_name] = None
+                # if function is given as a tuple, take the input parameters provided
+            if type(func_dict[stats_name]) is tuple:
+                func = func_dict[stats_name][0]
+                # check that key word arguments are provided as dictionary
+                if not type(func_dict[stats_name][1]) is dict:
+                    raise TypeError(
+                        "Tuple must contain dictionary with key word arguments for function."
+                    )
+                else:
+                    kwargs = func_dict[stats_name][1]
+                    # default needs to be sequence when function output is array-like
+                    output = func(np.random.rand(1, 10), **kwargs)
+                    if hasattr(output, "__len__"):
+                        default = np.full(output.shape, default)
+                    stats = np.array(
+                        [
+                            func(
+                                *[
+                                    field.ravel()[argsorted[bins[i - 1] : bins[i]]]
+                                    for field in fields
+                                ],
+                                **kwargs,
+                            )
+                            if bins[i] > bins[i - 1]
+                            else default
+                            for i in index
+                        ]
+                    )
+            # otherwise apply function on region without any input parameter
+            else:
+                func = func_dict[stats_name]
+                # default needs to be sequence when function output is array-like
+                output = func(np.random.rand(1, 10))
+                if hasattr(output, "__len__"):
+                    default = np.full(output.shape, default)
+
+                stats = np.array(
+                    [
+                        func(
+                            *[
+                                field.ravel()[argsorted[bins[i - 1] : bins[i]]]
+                                for field in fields
+                            ]
+                        )
+                        if bins[i] > bins[i - 1]
+                        else default
+                        for i in index
+                    ]
+                )
```

Comment:
We can avoid duplicating this code by using functools.partial to wrap the keywords with the function call.

e.g.: 
```
if type(func_dict[stats_name]) is tuple:
    func = partial(func_dict[stats_name][0], **func_dict[stats_name][1])
else:
    func = func_dict[stats_name][0]
```

Reply:
Nice idea, I have done that!